File: creator-suite-frontend/src/services/adminFeedback.ts
import { request } from 'umi';

export interface AdminFeedbackFilters {
  rating?: number;
  task_type?: string;
  service_id?: number;
  user_id?: number;
  organization_id?: number;
  date_from?: string;
  date_to?: string;
  has_text_feedback?: boolean;
}

export interface UserInfo {
  id: number;
  email: string;
  username: string;
  name?: string;
  organization_id?: number;
}

export interface ServiceInfo {
  id: number;
  name: string;
  description: string;
  cost_per_generation: number;
}

export interface CreationTaskInfo {
  id: string;
  task_type: string;
  provider: string;
  status: string;
  input_data: any;
  output_assets?: any;
  local_video_url?: string;
  local_image_url?: string;
  local_thumbnail_url?: string;
  error_message?: string;
  processing_time_seconds?: number;
  created_at: string;
  updated_at?: string;
  service?: ServiceInfo;
}

export interface AdminFeedbackDetail {
  id: number;
  rating: number;
  feedback_text?: string;
  created_at: string;
  updated_at?: string;
  user: UserInfo;
  creation_task: CreationTaskInfo;
}

export interface AdminFeedbackStats {
  total_feedbacks: number;
  average_rating: number;
  rating_distribution: {
    1: number;
    2: number;
    3: number;
    4: number;
    5: number;
  };
  feedbacks_by_task_type: Record<string, number>;
  feedbacks_by_service: Record<string, number>;
  recent_feedbacks_count: number;
}

export interface AdminFeedbackResponse {
  feedbacks: AdminFeedbackDetail[];
  total_count: number;
  page: number;
  page_size: number;
  total_pages: number;
  stats?: AdminFeedbackStats;
}

export interface AdminFeedbackListParams extends AdminFeedbackFilters {
  page?: number;
  page_size?: number;
  include_stats?: boolean;
}

/**
 * Get all feedbacks for admin view
 */
export async function getAdminFeedbacks(params?: AdminFeedbackListParams): Promise<AdminFeedbackResponse> {
  return request('/api/v1/admin/feedback/', {
    method: 'GET',
    params,
  });
}

/**
 * Get feedback statistics for admin dashboard
 */
export async function getAdminFeedbackStats(filters?: AdminFeedbackFilters): Promise<AdminFeedbackStats> {
  return request('/api/v1/admin/feedback/stats', {
    method: 'GET',
    params: filters,
  });
}

/**
 * Get detailed information about a specific feedback
 */
export async function getAdminFeedbackDetail(feedbackId: number): Promise<AdminFeedbackDetail> {
  return request(`/api/v1/admin/feedback/${feedbackId}`, {
    method: 'GET',
  });
}

/**
 * Delete a feedback (admin only)
 */
export async function deleteAdminFeedback(feedbackId: number): Promise<{ message: string }> {
  return request(`/api/v1/admin/feedback/${feedbackId}`, {
    method: 'DELETE',
  });
}

/**
 * Get feedback summary for a specific user
 */
export async function getUserFeedbackSummary(userId: number): Promise<{
  total_feedbacks: number;
  average_rating: number;
  rating_distribution: Record<number, number>;
}> {
  return request(`/api/v1/admin/feedback/user/${userId}/summary`, {
    method: 'GET',
  });
}

File: creator-suite-backend/app/schemas/admin_feedback.py
from datetime import datetime
from typing import Optional, List, Any
from pydantic import BaseModel

from app.schemas.feedback import Feedback


class UserInfo(BaseModel):
    """User information for admin feedback view"""
    id: int
    email: str
    username: str
    name: Optional[str] = None
    organization_id: Optional[int] = None

    model_config = {"from_attributes": True}


class ServiceInfo(BaseModel):
    """Service information for admin feedback view"""
    id: int
    name: str
    description: str
    cost_per_generation: float

    model_config = {"from_attributes": True}


class CreationTaskInfo(BaseModel):
    """Creation task information for admin feedback view"""
    id: str
    task_type: str
    provider: str
    status: str
    input_data: Any  # Can be dict or other JSON types
    output_assets: Optional[Any] = None  # Can be list of dicts or other JSON types
    local_video_url: Optional[str] = None
    local_image_url: Optional[str] = None
    local_thumbnail_url: Optional[str] = None
    error_message: Optional[str] = None
    processing_time_seconds: Optional[float] = None
    created_at: datetime
    updated_at: Optional[datetime] = None
    service: Optional[ServiceInfo] = None

    model_config = {"from_attributes": True}


class AdminFeedbackDetail(BaseModel):
    """Detailed feedback information for admin view"""
    id: int
    rating: int
    feedback_text: Optional[str] = None
    created_at: datetime
    updated_at: Optional[datetime] = None
    
    # Related information
    user: UserInfo
    creation_task: CreationTaskInfo

    model_config = {"from_attributes": True}


class AdminFeedbackStats(BaseModel):
    """Feedback statistics for admin dashboard"""
    total_feedbacks: int
    average_rating: float
    rating_distribution: dict
    feedbacks_by_task_type: dict
    feedbacks_by_service: dict
    recent_feedbacks_count: int  # Last 7 days


class AdminFeedbackFilters(BaseModel):
    """Filters for admin feedback listing"""
    rating: Optional[int] = None
    task_type: Optional[str] = None
    service_id: Optional[int] = None
    user_id: Optional[int] = None
    organization_id: Optional[int] = None
    date_from: Optional[datetime] = None
    date_to: Optional[datetime] = None
    has_text_feedback: Optional[bool] = None


class AdminFeedbackResponse(BaseModel):
    """Response model for admin feedback listing"""
    feedbacks: List[AdminFeedbackDetail]
    total_count: int
    page: int
    page_size: int
    total_pages: int
    stats: Optional[AdminFeedbackStats] = None
